sql/core/src/main/scala/org/apache/spark/sql/streaming/DataStreamReader.scala

```diff
@@ -64,8 +64,10 @@ final class DataStreamReader private[sql](sparkSession: SparkSession) extends Lo
    * @since 2.0.0
    */
   def schema(schema: StructType): DataStreamReader = {
-    val replaced = CharVarcharUtils.failIfHasCharVarchar(schema).asInstanceOf[StructType]
-    this.userSpecifiedSchema = Option(replaced)
+    if (schema != null) {
+      val replaced = CharVarcharUtils.failIfHasCharVarchar(schema).asInstanceOf[StructType]
+      this.userSpecifiedSchema = Option(replaced)
+    }
```

Comment:
Shall we change line 81~86 together?
- https://github.com/apache/spark/pull/31593/files#diff-f53c71f87464270c195a2aa190bf2d3310e48e4e3f48cd6f6be5ba43f32f1422R81-R86

Reply:
+1